diff -ur pktgen-orig/app/pktgen-constants.h pktgen/app/pktgen-constants.h
--- pktgen-orig/app/pktgen-constants.h	2017-01-11 16:07:16.422363435 +0100
+++ pktgen/app/pktgen-constants.h	2017-01-11 16:05:42.222365283 +0100
@@ -71,7 +71,7 @@
 #include <rte_mbuf.h>
 
 enum {
-	DEFAULT_PKT_BURST       = 32,	/* Increasing this number consumes memory very fast */
+	DEFAULT_PKT_BURST       = 256,	/* Increasing this number consumes memory very fast */
 #ifdef RTE_LIBRTE_VMXNET3_PMD
 	DEFAULT_RX_DESC         = (DEFAULT_PKT_BURST * 8 * 2),
 	DEFAULT_TX_DESC         = DEFAULT_RX_DESC,
diff -ur pktgen-orig/app/pktgen.h pktgen/app/pktgen.h
--- pktgen-orig/app/pktgen.h	2017-01-11 16:07:16.418363436 +0100
+++ pktgen/app/pktgen.h	2017-01-11 16:06:11.810364702 +0100
@@ -256,7 +256,7 @@
 	DEFAULT_DST_PORT        = 5678,
 	DEFAULT_PKT_NUMBER      = 0x012345678,
 	DEFAULT_ACK_NUMBER      = 0x012345690,
-	DEFAULT_WND_SIZE        = 8192,
+	DEFAULT_WND_SIZE        = 18192,
 	MIN_VLAN_ID             = 1,
 	MAX_VLAN_ID             = 4095,
 	DEFAULT_VLAN_ID         = MIN_VLAN_ID,
diff -ur pktgen-orig/lib/archive/lua-5.3.0/src/luaconf.h pktgen/lib/archive/lua-5.3.0/src/luaconf.h
--- pktgen-orig/lib/archive/lua-5.3.0/src/luaconf.h	2017-01-11 16:07:16.394363436 +0100
+++ pktgen/lib/archive/lua-5.3.0/src/luaconf.h	2017-01-11 16:09:34.646360725 +0100
@@ -704,7 +704,7 @@
 @@ LUAL_BUFFERSIZE is the buffer size used by the lauxlib buffer system.
 ** CHANGE it if it uses too much C-stack space.
 */
-#define LUAL_BUFFERSIZE	((int)(0x80 * sizeof(void*) * sizeof(lua_Integer)))
+#define LUAL_BUFFERSIZE	((int)(0x200 * sizeof(void*) * sizeof(lua_Integer)))
 
 /* }================================================================== */
 
diff -ur pktgen-orig/lib/lua/src/luaconf.h pktgen/lib/lua/src/luaconf.h
--- pktgen-orig/lib/lua/src/luaconf.h	2017-01-11 16:10:52.386359201 +0100
+++ pktgen/lib/lua/src/luaconf.h	2017-01-11 16:05:06.430365984 +0100
@@ -733,9 +733,9 @@
 ** 'string.format'.)
 */
 #if defined(LUA_FLOAT_LONGDOUBLE)
-#define LUAL_BUFFERSIZE		8192
+#define LUAL_BUFFERSIZE		65536
 #else
-#define LUAL_BUFFERSIZE   ((int)(0x80 * sizeof(void*) * sizeof(lua_Integer)))
+#define LUAL_BUFFERSIZE   ((int)(0x200 * sizeof(void*) * sizeof(lua_Integer)))
 #endif
 
 /* }================================================================== */
