diff --git a/lib/librte_eal/linuxapp/eal/eal_log.c b/lib/librte_eal/linuxapp/eal/eal_log.c
index c088bd9..11e3c5c 100644
--- a/lib/librte_eal/linuxapp/eal/eal_log.c
+++ b/lib/librte_eal/linuxapp/eal/eal_log.c
@@ -88,6 +88,9 @@ static cookie_io_functions_t console_log_func = {
 int
 rte_eal_log_init(const char *id, int facility)
 {
+#ifdef KLEE_VERIFICATION
+	FILE *log_stream = NULL; // use stderr for logging, we don't support syslog...
+#else
 	FILE *log_stream;
 
 	log_stream = fopencookie(NULL, "w+", console_log_func);
@@ -95,6 +98,7 @@ rte_eal_log_init(const char *id, int facility)
 		return -1;
 
 	openlog(id, LOG_NDELAY | LOG_PID, facility);
+#endif
 
 	eal_log_set_default(log_stream);
