diff --git a/drivers/net/ixgbe/ixgbe_ethdev.c b/drivers/net/ixgbe/ixgbe_ethdev.c
index ff19a56..b1b39eb 100644
--- a/drivers/net/ixgbe/ixgbe_ethdev.c
+++ b/drivers/net/ixgbe/ixgbe_ethdev.c
@@ -1107,7 +1107,7 @@ ixgbe_swfw_lock_reset(struct ixgbe_hw *hw)
 	 * lock can not be taken it is due to an improper lock of the
 	 * semaphore.
 	 */
-	mask = IXGBE_GSSR_EEP_SM | IXGBE_GSSR_MAC_CSR_SM | IXGBE_GSSR_SW_MNG_SM;
+	mask = IXGBE_GSSR_EEP_SM | IXGBE_GSSR_MAC_CSR_SM;
 	if (ixgbe_acquire_swfw_semaphore(hw, mask) < 0) {
 		PMD_DRV_LOG(DEBUG, "SWFW common locks released");
 	}
