diff --git a/lib/librte_eal/common/include/arch/x86/rte_cycles.h b/lib/librte_eal/common/include/arch/x86/rte_cycles.h
index 6e3c7d8..17d6617 100644
--- a/lib/librte_eal/common/include/arch/x86/rte_cycles.h
+++ b/lib/librte_eal/common/include/arch/x86/rte_cycles.h
@@ -98,10 +98,14 @@ rte_rdtsc(void)
 	}
 #endif
 
+#ifdef KLEE_VERIFICATION
+	return 0; // inline assembly? no way!
+#else
 	asm volatile("rdtsc" :
 		     "=a" (tsc.lo_32),
 		     "=d" (tsc.hi_32));
 	return tsc.tsc_64;
+#endif
 }
 
 static inline uint64_t
diff --git a/lib/librte_eal/linuxapp/eal/eal_interrupts.c b/lib/librte_eal/linuxapp/eal/eal_interrupts.c
index 47a3b20..94679e1 100644
--- a/lib/librte_eal/linuxapp/eal/eal_interrupts.c
+++ b/lib/librte_eal/linuxapp/eal/eal_interrupts.c
@@ -862,6 +862,10 @@ eal_intr_thread_main(__rte_unused void *arg)
 int
 rte_eal_intr_init(void)
 {
+#ifdef KLEE_VERIFICATION
+	// we don't support pipes
+	return 0;
+#endif
 	int ret = 0, ret_1 = 0;
 	char thread_name[RTE_MAX_THREAD_NAME_LEN]; 

diff --git a/lib/librte_eal/linuxapp/eal/eal_log.c b/lib/librte_eal/linuxapp/eal/eal_log.c
index c088bd9..11e3c5c 100644
--- a/lib/librte_eal/linuxapp/eal/eal_log.c
+++ b/lib/librte_eal/linuxapp/eal/eal_log.c
@@ -88,6 +88,9 @@ static cookie_io_functions_t console_log_func = {
 int
 rte_eal_log_init(const char *id, int facility)
 {
+#ifdef KLEE_VERIFICATION
+	FILE *log_stream = NULL; // use stderr for logging, we don't support syslog...
+#else
 	FILE *log_stream;
 
 	log_stream = fopencookie(NULL, "w+", console_log_func);
@@ -95,6 +98,7 @@ rte_eal_log_init(const char *id, int facility)
 		return -1;
 
 	openlog(id, LOG_NDELAY | LOG_PID, facility);
+#endif
 
 	eal_log_set_default(log_stream);
